test/functional/blocknotify.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test blocknotify.
+
+Verify that a bitcoind node calls the given command for each block.
+"""
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, wait_until
+
+class BlockNotifyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.path = os.path.join(self.options.tmpdir, 'blocks.txt')
+
+        # the command will append the notified block hash to file
+        self.extra_args = [['-blocknotify=echo %%s >> %s' % self.path]]
+        super().setup_network()
+
+    def run_test(self):
+        block_count = 10
+        blocks = self.nodes[0].generate(block_count)
+
+        wait_until(lambda: os.path.isfile(self.path), timeout=1)
+        # wait at most 10 seconds for expected file size before reading the content
+        wait_until(lambda: os.stat(self.path).st_size >= (block_count * 65), timeout=10)
+
+        # file content should equal the generated blocks hashes
+        with open(self.path, 'r') as f:
+            assert_equal(blocks, f.read().splitlines())
```

Comment:
Hmm. I don't think boost::thread guarantees that the thread run in the order they are constructed. Am I wrong?

The arrays should have the same content, but a different order.

Reply:
Correct, in `-walletnotify` (#10966) there's:
```cpp
            # txids are sorted because there is no order guarantee in notifications
            assert_equal(sorted(txids_rpc), sorted(txids_file))
```
So I'll do the same here for correctness.